### ISSUE TITLE
Add a `RELEASE.md` descrihing how to release a new version of CF

### ISSUE THREAD

**Issue by davidhassell** [cf-convention/cf-conventions#371]
After CF-1.9 was released the release procedure was written up, but never made to a PR. Here it is now, in time for CF-1.10.

**Comment by erget** [comment]
Hi @davidhassell I made some changes and also inserted some comments - have a look and see what you think :)

**Comment by erget** [comment]
Additional changes concerning keeping contributor list up to date agreed in https://github.com/cf-convention/cf-convention.github.io/issues/230. These were introduced in 0b20db2572342628b954960a4a777eefd66efe74 and bf274bcf2165b20b38061b0c5f48ecb2cc6f08c7

**Comment by JonathanGregory** [comment]
In `CONTRIBUTING.md` we have

> When an issue and accompanying pull request concludes with an agreed change to the conventions, the pull request is merged into the master branch according to the Rules for CF Convention Changes.
> The pull request is merged by a member of the Governance Panel or the CF Conventions and Standard Names Committee. The person who merges the pull request also closes the issue.

Following https://github.com/cf-convention/cf-conventions/issues/345 agreed last week, we should add here:

> The issue should be given the `change agreed` label, and the pull request should be given the next release number as a milestone.

I suggest that this sentence or something like it should be inserted between the other two.

Jonathan

**Comment by erget** [comment]
@JonathanGregory that looks fine for me, I would approve that.

BTW your note reminds me that we have yet to align the terminology everywhere of using "main" rather than "master" as branch name; I've submitted https://github.com/cf-convention/cf-conventions/pull/376 to correct this in `CONTRIBUTING.md`, somehow I missed that when implementing https://github.com/cf-convention/discuss/issues/125. Would appreciate if somebody takes a look at those minor but important changes.

**Comment by JonathanGregory** [comment]
I have inserted a list of tasks, because I thought that made it clearer. Please see [this PR](https://github.com/cf-convention/cf-conventions/pull/377/files).

**Comment by erget** [comment]
Now this issue has produced 3 PRs. One's already merged because it was a simple error. Are the other 2 agreed? If they are, we could merge them sequentially, or merge 1 into the other and then merge the whole shebang when that's done. Both open PRs are valuable in my view and should be merged once it's clear that they're agreed (from my perspective that could be the case now, but others may disagree).

I believe the cleanest thing would be

- [x] Raise a PR from branch [JonathanGregory:issue345+371](https://github.com/JonathanGregory/cf-conventions/tree/issue345%2B371) to [davidhassell:release-procedure](https://github.com/davidhassell/cf-conventions/tree/release-procedure). That would put those changes together with those already discussed in #372 once merged. I've done this now; @davidhassell if you're happy with those changes please merge that PR.
- [ ] Close #377, since those changes are now in #372
- [ ] Review #372 in its entirety, then merge once agreed.

Does that make sense?

**Comment by davidhassell** [comment]
Sounds good.Just to be clear, "that PR" of the first action is #377 ?

**Comment by davidhassell** [comment]
(premature send) .... also, might be good to note that we only intend to merge #372 after we've used it for 1.10, as only then do we know it works (https://github.com/cf-convention/cf-conventions/pull/372#issuecomment-1220786212)

**Comment by erget** [comment]
> Sounds good.Just to be clear, "that PR" of the first action is #377 ?

Yes.

> (premature send) .... also, might be good to note that we only intend to merge https://github.com/cf-convention/cf-conventions/pull/372 after we've used it for 1.10, as only then do we know it works (https://github.com/cf-convention/cf-conventions/pull/372#issuecomment-1220786212)

I'd forgotten about that, absolutely right.

**Comment by larsbarring** [comment]
Hi @davidhassell I just had quick look at your new PR #498. From my outside perspective it looks good, but I guess only you and other release manages can judge.  Nevertheless, here are two small observations:
* The release date, that I tried to automate in [issue #457](https://github.com/cf-convention/cf-conventions/issues/457) / [PR#475](https://github.com/cf-convention/cf-conventions/pull/475) didn't work out in the end (hence [cf-convention.github.io/428](https://github.com/cf-convention/cf-convention.github.io/pull/428)) I am not sure whether this automation could be fixed, or if the manual step needs to be recorded in the procedure.
* The released *version number*, and the new *draft version number*, appears at quite a few places. Is this something that could be further automated ?

**Comment by JonathanGregory** [comment]
Once a release is done, we need a milestone for the next one. I have just created the 1.12 milestone in order to attribute a PR to it.

**Comment by davidhassell** [comment]
Thanks, @JonathanGregory, creating the next milestone has been added (https://github.com/cf-convention/cf-conventions/pull/498/commits/0d773294e4b109e131b455b89b9b374650c5e65a).

**Comment by davidhassell** [comment]
Hi. Yet another PR for the release procedure: #588 

This was used for the release of CF-1.12, and is the simplest and mot robust one yet - mainly due to the great effort of others to make this a nicer process - thank you :)

The procedure described in the PR is what I actually did for releasing CF-1.12, so I guess that's a proof that it works. It would be nice if someone could look through it and imagine they were doing it for themselves, to see if it makes sense!

It might be a good idea to add a fifth checkbox to the Pull Request template to ask if the release procedure document is up to date. Hopefully in that way it will evolve nicely.

David

**Comment by JonathanGregory** [comment]
Dear David

Thank you! It looks good to me. I have noted a couple of typos. Also, I have suggestions:

* Having added new contributors to the list of contributors, remove the `new contributor` label from the issue concerned.
* Add the `change agreed` label to any issue which has been incorporated in the new release, if it's not there.
* Close the issue, if it isn't already.
* Something about milestones, perhaps. It must be a preliminary step that you make sure all the accepted changes to be included have the milestone for the new release. You and I discussed this, but I don't recall what we said. Should the milestone be attached to the issue as well as the PR, for instance?

Best wishes

Jonathan

**Comment by davidhassell** [comment]
Dear @JonathanGregory,

Having just said we should merge this, I realise that I missed [your suggestions](https://github.com/cf-convention/cf-conventions/issues/371#issuecomment-2610607716) - sorry! I have incorporated your ideas, including a note on inserting the pre-reserved DOI: 42d6815e1fce790cac2990ad47a9db93f26d02b7

**Comment by JonathanGregory** [comment]
Thanks, @davidhassell. I've added a couple of inline comments to the PR.

**Comment by JonathanGregory** [comment]
Dear David

Could we conclude and merge this? I think everyone's happy with it, except that I made some small inline comments to your PR #588. What do you think of those?

Thanks

Jonathan

**Comment by larsbarring** [comment]
Ready to close now? Else candidate for the housekeeping session at CF2026  Workshop.

**Comment by JonathanGregory** [comment]
I believe @davidhassell is on leave at the moment, and I hope he will have time to merge this when he returns.

**Comment by davidhassell** [comment]
Hello - I've resolved @JonathanGregory's comments (they were all pertinent) - good to merge, from my perspective!
Thanks,
David

**Comment by JonathanGregory** [comment]
Thanks, David. I will merge it now.
